### ISSUE TITLE
Slow responses while host bind to 0.0.0.0

### ISSUE THREAD

**Issue by agozuberk** [jolokia/jolokia#461]
Hi,

We use Jolokia JVM agent 1.6.2 for our Quarkus based microservices on Openshift. It is intended to use Jolokia with Hawtio in order to monitor the our applications. 

**Summary:** 

If I run the agent with "host=localhost" option, Jolokia is working without any issue. But Hawtio is not able to access to it, because 8778 port is listen on localhost. If I specify host as "0.0.0.0", we are able access the Jolokia with pod ip, but query time is so long - between 30-70 seconds. Hawtio get timeouts.

**Details:** 

I. Bind localhost 
I run the agent with the following configuration:

`-javaagent:/tmp/jolokia/jolokia-jvm-1.6.2-agent.jar=debug=true,host=localhost,allowDnsReverseLookup=false`

It is accessible wit localhost and it returns quickly.

![image](https://user-images.githubusercontent.com/12867001/117648979-9bfa9200-b197-11eb-8731-5a72ae0bb329.png)

II. Bind 0.0.0.0
I set the options as following:

`-javaagent:/tmp/jolokia/jolokia-jvm-1.6.2-agent.jar=debug=true,host=0.0.0.0,allowDnsReverseLookup=false`

Now I am able to access with pod ip too, but query time is too long.

![image](https://user-images.githubusercontent.com/12867001/117649103-c8161300-b197-11eb-8e8a-dbf2069e00bc.png)

I open the debug level logging and set the "allowDnsReverseLookup" option false as suggested in this issue, but it is not working: 
https://github.com/rhuss/jolokia/issues/256#issuecomment-215411388

Also I suspect any network issue and try to use localhost within the pod while host bound on 0.0.0.0 but there is no difference. 
![image](https://user-images.githubusercontent.com/12867001/117650484-6ce52000-b199-11eb-8a4b-053358e8e3d8.png)

Any help is appreciated.

**Comment by rhuss** [comment]
You could try to create a thread dump while the request is running to detect which thread is waiting on something. I still expect it to be a DNS issue (it's always DNS ;-). Unfortunately, I'm not of much help here, as I'm totally overloaded with my day-to-day work, and helping with Jolokia is on a very low priority (I'm passing by only every 1 or 2 months). Sorry for that.

**Comment by grgrzybek** [comment]
Jolokia doesn't control the networking layer it runs on. This looks like an OpenShift/K8S/routing/DNS problem.
I'm closing for now.
